c++/src/Reader.cc

```diff
@@ -290,6 +291,15 @@ namespace orc {
     return selectedColumns;
   }
 
+  ArrayReadIntent RowReaderImpl::getReadIntent(uint64_t typeId) const {
+    auto elem = readIntents.find(typeId);
+    if (elem == readIntents.end()) {
+      return ArrayReadIntent_ALL;
+    } else {
+      return elem->second;
```

Comment:
Do we have a test coverage for this code path?

Reply:
The TestColumnReaderEncoded.testListOffsetsOnly that I add only mock up to StripeStreamsImpl::getReadIntent(), and I believe it didn't touch this part.

I will try to add more comprehensive tests. Do you know any existing unit tests I can refer to?

Reply:
Created 4 new test:

- TestReadIntent.testListAll
- TestReadIntent.testListOffsets
- TestReadIntent.testListAllAndOffsets
- TestReadIntent.testListConflictingIntent